sample-app/src/utils/geolocationutils.tsx

```diff
@@ -0,0 +1,56 @@
+import { AnswersActions } from "@yext/answers-headless-react";
+import { SearchIntent } from "@yext/answers-headless";
+
+const defaultGeolocationOptions: PositionOptions = {
+  enableHighAccuracy: false,
+  timeout: 6000,
+  maximumAge: 300000,
+};
+
+/**
+ * Executes universal or vertical search with user location retrieved from nagivator.geolocation API.
+ * If useLocationOnNearMeOnly is true, will attempt to get user location only if the query's searchIntents
+ * (retrieved through autocomplete response) contains 'NEAR_ME'. If there's a failure in getting
+ * user location, a normal search is performed.
+ */
+export async function executeSearchWithUserLocation(
+  answersActions: AnswersActions,
+  isVertical: boolean,
+  geolocationOptions?: PositionOptions,
+  useLocationOnNearMeOnly?: boolean
```

Comment:
I think we should consider renaming this method. The method name suggests we always provide the user's location. But, if you dig deeper, you see there's a param that lets us search without sending the user's location.

Reply:
update: split into functions inside `SearchHandler` class as suggested in another comment